ledger/accountdb.go

```diff
@@ -1989,6 +1995,43 @@ func (qs *accountsDbQueries) lookupKeyValue(key string) (pv persistedValue, err
 	return
 }
 
+func (qs *accountsDbQueries) lookupKeysByPrefix(
+	prefix string, maxKeyNum uint64, sawDeletedKeys map[string]bool, keysWithValue *map[string]bool,
+) (round basics.Round, err error) {
+	err = db.Retry(func() (_err error) {
+		rows, _err := qs.lookupKeysByPrefixStmt.Query(prefix)
+		if _err != nil {
+			if _err == sql.ErrNoRows {
+				_err = fmt.Errorf("unable to query value for prefix %v : %w", prefix, _err)
+				return
+			}
+		}
+		defer rows.Close()
+
+		var keyName string
+
+		for rows.Next() {
+			if maxKeyNum > 0 && len(*keysWithValue) == int(maxKeyNum) {
+				break
+			}
+			_err = rows.Scan(&round, &keyName)
+			if _err != nil {
+				return
+			}
+			// I assume that in DB, there is only key with valid value (no empty value)
+			if _, ok := sawDeletedKeys[keyName]; ok {
+				continue
+			}
+			if _, ok := (*keysWithValue)[keyName]; ok {
+				continue
+			}
+			(*keysWithValue)[keyName] = true
```

Comment:
Also simplified, if you just pass in the results map.
```suggestion
			if _, ok := results[keyName]; ok {
				continue
			}
			results[keyName] = true
```

Reply:
good point on `LIMIT` in sql query, I will update the limit number in the query.

Reply:
updated in latest commit https://github.com/algorand/go-algorand/pull/4122/commits/df6bfac917f8b1068a3b1deb03559a8db9b58ff1

Reply:
After more thoughts about it, I decide to revert the limit in sql query. The problem of using `LIMIT` in sql query is that, it does not always return all the valid result we want, probably partly. Some of its query result might already been seen, whether it is set or delete.